src/gwt/src/org/rstudio/core/client/VirtualConsole.java

```diff
@@ -374,12 +377,15 @@ else if (start <= l && end <= r && end > l)
                // reduce the original range and add ours
                overlap.trimLeft(delta);
 
-               if (!range.text().isEmpty())
-                  insertions.add(range);
-
                // move the shortened range to its new start position
-               moves.put(l, overlap.start);
+               // unless it's empty
+               if (overlap.length > 0) {
```

Comment:
Formatting, our standard for Java code is opening brace on its own line.

https://github.com/rstudio/rstudio/wiki/Coding-Standards-(java,-cpp)